uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/OrchestratorZoneControllerMockMvcTests.java

```diff
@@ -375,6 +373,91 @@ void testCreateZone_Accepted_Success() throws Exception {
             .andExpect(status().isAccepted()).andReturn();
     }
 
+    @Test
+    void testCreateZone_MessageNotReadable_InvalidFormatError() throws Exception {
+        MvcResult result = mockMvc
+            .perform(
+                post("/orchestrator/zones")
+                    .contentType(APPLICATION_JSON)
+                    .content("[[[[ ]]]]")
+                    .header("Authorization", "Bearer " + uaaAdminClientToken))
+            .andExpect(status().isBadRequest()).andReturn();
+
+        assertEquals(APPLICATION_JSON_VALUE, result.getResponse().getContentType());
+        assertTrue(result.getResponse().getContentAsString().contains("Request failed due to a validation error"));
+    }
+
+    @Test
+    void testCreateZone_MessageNotReadable_JsonMappingException() throws Exception {
+        MvcResult result = mockMvc
+            .perform(
+                post("/orchestrator/zones")
+                    .contentType(APPLICATION_JSON)
+                    .content("{\n" +
+                             "  \"name\": \"tes5000-00\",\n" +
+                             "  \"parameters\": {\n" +
+                             "    \"adminClientSecret\": 0992932.303203.00223\n" +
+                             "    \"subdomain\" : \"uywyyw\"\n" +
+                             "  }\n" +
+                             "}")
+                    .header("Authorization", "Bearer " + uaaAdminClientToken))
+            .andExpect(status().isBadRequest()).andReturn();
+
+        assertEquals(APPLICATION_JSON_VALUE, result.getResponse().getContentType());
+        assertTrue(result.getResponse().getContentAsString().contains("parameters Invalid numeric value"));
+    }
+
+    @Test
+    void testCreateZone_MessageNotReadable_MismatchedInputException() throws Exception {
+        MvcResult result = mockMvc
+            .perform(
+                post("/orchestrator/zones")
+                    .contentType(APPLICATION_JSON)
+                    .content("{\n" +
+                             "  \"name\": [\"323231\", \"323232\", \"323233\"],\n" +
+                             "  \"parameters\": {\n" +
+                             "    \"adminClientSecret\": \"dsfds\",\n" +
+                             "    \"subdomain\" : \"test-zone-0\"\n" +
+                             "  }\n" +
+                             "}")
+                    .header("Authorization", "Bearer " + uaaAdminClientToken))
+            .andExpect(status().isBadRequest()).andReturn();
+
+        assertEquals(APPLICATION_JSON_VALUE, result.getResponse().getContentType());
+        assertTrue(result.getResponse().getContentAsString().contains("name Cannot deserialize value of type " +
```

Comment:
Same casing thing here with `Cannot` word.

Reply:
Done.

Reply:
As discussed over daily scrum call, removed "Character.toLowerCase(errorMessage.charAt(0)) + errorMessage.substring(1) from the code and added " is invalid: " string